partiql-parser/src/result.rs

```diff
@@ -190,6 +190,10 @@ pub enum ParserError {
     /// Indicates that there is a problem with run-time violation of some API.
     #[error("Invalid Argument: {message}")]
     InvalidArgument { message: String },
+
+    /// Indicates that there is an internal error that isn't SyntaxError nor InvalidArgument.
+    #[error("Illegal State")]
+    IllegalState,
```

Comment:
I think following the pattern above to have a `message` field would be a good idea.

I do think in the future we'll make this a bit more factored out as part of #18, but for now the slightly duplicative `{ message: String }` pattern is reasonable.